Sources/Parsing/Parsers/Float.swift

```diff
@@ -0,0 +1,172 @@
+  /// - Returns: A parser that consumes a floating-point number from the beginning of a collection
+  ///   of UTF-8 code units.
+  @inlinable
+  public static func parser<Input>(
+    of inputType: Input.Type = Input.self
+  ) -> Parsers.FloatParser<Input, Self> {
+    .init()
+  }
+
+  /// A parser that consumes a floating-point number from the beginning of a substring's UTF-8 view.
+  ///
+  /// This overload is provided to allow the `Input` generic to be inferred when it is
+  /// `Substring.UTF8View`.
+  ///
+  /// See <doc:Double> for more information about this parser.
+  ///
+  /// - Parameter inputType: The `Substring.UTF8View` type. This parameter is included to mirror the
+  ///   interface that parses any collection of UTF-8 code units.
+  /// - Returns: A parser that consumes a floating-point number from the beginning of a substring's
+  ///   UTF-8 view.
+  @_disfavoredOverload
+  @inlinable
+  public static func parser(
+    of inputType: Substring.UTF8View.Type = Substring.UTF8View.self
+  ) -> Parsers.FloatParser<Substring.UTF8View, Self> {
+    .init()
+  }
+
+  /// A parser that consumes a floating-point number from the beginning of a substring.
+  ///
+  /// This overload is provided to allow the `Input` generic to be inferred when it is `Substring`.
+  ///
+  /// See <doc:Double> for more information about this parser.
+  ///
+  /// - Parameter inputType: The `Substring` type. This parameter is included to mirror the
+  ///   interface that parses any collection of UTF-8 code units.
+  /// - Returns: A parser that consumes a floating-point number from the beginning of a substring.
+  @_disfavoredOverload
+  @inlinable
+  public static func parser(
+    of inputType: Substring.Type = Substring.self
+  ) -> FromUTF8View<Substring, Parsers.FloatParser<Substring.UTF8View, Self>> {
+    .init { Parsers.FloatParser<Substring.UTF8View, Self>() }
+  }
+}
+
+extension Parsers {
+  /// A parser that consumes a floating-point number from the beginning of a collection of UTF-8
+  /// code units.
+  ///
+  /// You will not typically need to interact with this type directly. Instead you will usually use
+  /// the static `parser()` method on the `BinaryFloatingPoint` of your choice, e.g.,
+  /// `Double.parser()`, `Float80.parser()`, etc., all of which construct this type.
+  ///
+  /// See <doc:Double> for more information about this parser.
+  public struct FloatParser<Input: Collection, Output: BinaryFloatingPoint>: Parser
+  where
+    Input.SubSequence == Input,
+    Input.Element == UTF8.CodeUnit,
+    Output: LosslessStringConvertible
+  {
+    @inlinable
+    public init() {}
+
+    @inlinable
+    public func parse(_ input: inout Input) throws -> Output {
+      let original = input
+      let s = input.parseFloat()
+      guard let n = Output(String(decoding: s, as: UTF8.self))
+      else {
+        throw ParsingError.expectedInput("\(Output.self)".lowercased(), from: original, to: input)
```

Comment:
Errors have changed slightly: `expected double` and `expected float` are unchanged, but if you use `Float80` you'll get `expected float80` now.